cloud/amazon/emr.py

```diff
@@ -0,0 +1,414 @@
+                                              aws_access_key_id=self.aws_access_key,
+                                              aws_secret_access_key=self.aws_secret_key)
+            return(conn)
+
+        except boto.exception.NoAuthHandlerFound, e:
+            self.module.fail_json(msg="Unable to create connection: {0}".format(str(e)))
+
+    def set_jobflow_instance_groups(self):
+        """
+        :param module: AnsibleModule object
+        :returns list: list of boto.emr.InstanceGroup objects
+        """
+
+        jobflow_instance_groups = []
+
+        for ig in self.instance_groups:
+            group_type = ig.get('group_type', None)
+            instance_type = ig.get('instance_type', None)
+            instance_count = ig.get('count', 1)
+            my_spot = self.module.boolean(ig.get('spot'))
+            my_bidprice = ig.get('bidprice', None)
+
+            if int(instance_count) <= 0 or instance_type is None:
+                continue
+
+            if my_spot:
+                my_market='SPOT'
+                if not my_bidprice:
+                    self.module.fail_json(msg='SPOT instances require a bidprice')
+            else:
+                my_market='ON_DEMAND'
+
+            jobflow_instance_groups.append(InstanceGroup(int(instance_count),
+                                                         group_type.upper(),
+                                                         instance_type,
+                                                         my_market,
+                                                         "{0}_GROUP".format(group_type.upper()),
+                                                         bidprice=my_bidprice))
+        return(jobflow_instance_groups)
+
+    def set_jobflow_bootstrap_actions(self):
+        """
+        :param module: AnsibleModule object
+        :returns list: list of boto.emr.emrobject.BootstrapAction objects
+        """
+
+        my_bootstrap_actions = []
+        for bootstrap in self.bootstrap_actions:
+            args = []
+            raw_args = bootstrap.get('args', None)
+            if raw_args:
+                args = [x.strip() for x in raw_args.split(',')]
+
+            my_bootstrap_actions.append(BootstrapAction(bootstrap['name'],
+                                                        bootstrap['action'],
+                                                        args))
+        return(my_bootstrap_actions)
+
+    def set_jobflow_jar_steps(self):
+        """
+        :returns list: list of boto.emr.step.JarStep
+        """
+
+        my_jar_steps = []
+        for jar_step in self.jar_steps:
+            args = []
+            raw_args = jar_step.get('args', None)
+            if raw_args:
+                args = [x.strip() for x in raw_args.split(',')]
+
+            my_jar_steps.append(JarStep(jar_step['name'],
+                                        jar_step['jar_uri'],
+                                        None,
```

Comment:
we should add a jar_step['main'] instead of None, there's a possibility we want to run a main class
We can do something like this:
```
if jar_step['main'] == 'None':
   jar_step['main'] = None

my_jar_steps.append(JarStep(jar_step['name'],
                                        jar_step['jar_uri'],
                                        jar_step['main'],
                                        jar_step['action_on_failure'],
                                        args))

```